src/vulkan/buffer.h

```diff
@@ -22,42 +22,38 @@
 namespace amber {
 namespace vulkan {
 
+// Class managing Vulkan Buffer i.e., VkBuffer |buffer_|. |memory_|
+// has VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT and
+// VK_MEMORY_PROPERTY_HOST_COHERENT_BIT properties and it is mapped
+// to |buffer_|.
 class Buffer : public Resource {
  public:
   Buffer(VkDevice device,
          size_t size_in_bytes,
          const VkPhysicalDeviceMemoryProperties& properties);
   ~Buffer() override;
 
+  // Create |buffer_| whose usage is |usage| and allocate |memory_|
+  // with VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT and
+  // VK_MEMORY_PROPERTY_HOST_COHERENT_BIT properties. It also maps
+  // |memory_| to |buffer_|
   Result Initialize(const VkBufferUsageFlags usage);
+
   VkBuffer GetVkBuffer() const { return buffer_; }
   Result CreateVkBufferView(VkFormat format);
   VkBufferView GetVkBufferView() const { return view_; }
 
-  // If this buffer is host accessible and has coherent memory, this
-  // method does nothing. If this buffer is host accessible but it does
-  // not have coherent memory, this method flush the memory to make the
-  // writes from host effective to device. Otherwise, this method records
-  // the command for copying the secondary host-accessible buffer to
-  // this device local buffer. Note that it only records the command and
-  // the actual submission must be done later.
+  // Since |buffer_| is mapped to host accessible and host coherent
+  // memory |memory_|, this method only conducts memory barrier to
+  // make it available to device domain.
   Result CopyToDevice(VkCommandBuffer command);
 
   // Resource
-  VkDeviceMemory GetHostAccessMemory() const override {
-    if (is_buffer_host_accessible_)
-      return memory_;
-
-    return Resource::GetHostAccessMemory();
```

Comment:
Is Resource::GetHostAccessMemory still used or can it be removed?

Reply:
`Resource::GetHostAccessMemory()` is still used for image resource. It should not be removed.